docs/articles/kafka-record-headers.md

```
@@ -0,0 +1,140 @@
+<dependency>
+    <groupId>io.github.kzmlabs.flinkstatefun</groupId>
+    <artifactId>statefun-sdk-java</artifactId>
+    <version>3.4.0-KZM-3.4</version>
+</dependency>
+```
+
+Existing deployments upgrade in place — the protocol change is strictly additive, and header forwarding stays off until a topic opts in, so nothing changes behavior until you say so.
+
+Full reference: **[the Kafka record headers guide](https://kzmlabs.github.io/flink-statefun/guides/kafka-headers/)**. Background on the fork itself: **[We forked Apache Stateful Functions for Flink 2.x — here's why](forking-statefun.md)**.
+
+[our-repo]: https://github.com/kzmlabs/flink-statefun
```

markdownlint-cli2 (0.23.0)

[warning] Link and image reference definitions should be needed
Unused link or image reference definition: "our-repo"

(MD053, link-image-reference-definitions)